src/read.c

```diff
@@ -702,9 +709,6 @@ static avifResult avifDecoderReadItem(avifDecoder * decoder, avifDecoderItem * i
             if (readResult != AVIF_RESULT_OK) {
                 return readResult;
             }
```

Comment:
At this point, we have just called decoder->io->read(decoder->io, 0, extent->offset, bytesToRead, &offsetBuffer)  successfully. Therefore we know bytesToRead >= offsetBuffer.size. That implies the check at line 705 is equivalent to the check at line 710.